### DISCUSSION TITLE
how to set data to a var?

### DISCUSSION THREAD

**Discussion by dtlevin** [11ty/eleventy#2366] [Q&A]
I now have data: landingpage.json, and to display what I need, I am using {{ landingpage.hero.title }} to add the content to different components on the page, so I also have {{ landingpage.cta.title }} etc. I want to set the landingpage.json to a var name si I can point it to different json files for translation and not have to rest anything. So something like {{% set content = landingpage.json % }} and then in my components I can use {{ content.hero.title }}

**Comment by pdehaan** [comment]
@dtlevin Here's one approach:

```sh
tree -aI node_modules

.
├── .eleventy.js
├── package-lock.json
├── package.json
├── src
│   ├── _data
│   │   └── landingpage
│   │       ├── en.json
│   │       └── fr.json
│   └── landingpage
│       ├── en.njk
│       └── fr.njk
└── www
    └── landingpage
        ├── en
        │   └── index.html
        └── fr
            └── index.html

8 directories, 9 files
```

Instead of _data/landingpage.json, I created a _data/landingpage/en.json and _data/landingpage/fr.json with my localized content:

```json
{
  "hero": {
    "title": "Hello World"
  },
  "cta": {
    "title": "Click here"
  }
}
```
and
```json
{
  "hero": {
    "title": "Bonjour le monde"
  },
  "cta": {
    "title": "Cliquez ici"
  }
}
```

Then I created a landingpage/en.njk file with the following:

```njk
---
locale: en
---

{%- set landing = landingpage[locale] -%}

<h1>{{ landing.hero.title }}</h1>

<button>{{ landing.cta.title }}</button>
```

And a landingpage/fr.njk file with the following:
```njk
---
locale: fr
---

{%- set landing = landingpage[locale] -%}

<h1>{{ landing.hero.title }}</h1>

<button>{{ landing.cta.title }}</button>
```

As you can see, the only thing different is the `locale` front matter.
You could even invert it slightly to have a /en/landingpage.njk file instead and then create a directory data file like /en/en.json or /en/en.11tydata.js which sets the `locale: "en"` front matter for the entire directory.

And since my simplified "en" and "fr" templates are almost exactly the same (except for that front matter value) you could even create an include file or custom layout to have a bit less duplicated content.

**Comment by pdehaan** [comment]
Had a slightly different/better approach using pagination. https://github.com/pdehaan/11ty-2366/

Now my src/landingpage.njk looks like this:

```njk
---
layout: landingpage
---
```

And all the logic is kept in that layout file, src/_includes/layouts/landingpage.njk:
```njk
---
layout: base
pagination:
  data: landingpage
  size: 1
  alias: t
  resolve: values
permalink: "/{% if t.locale != 'en' %}{{ t.locale | slugify }}/{% endif %}"
eleventyComputed:
  locale: "{{ t.locale }}"
  title: "{{ t.hero.title }}"
---

<h1>{{ t.hero.title }}</h1>

<button>{{ t.cta.title }}</button>
```

FIrst we extend a src/_includes/layouts/base.njk file, then we paginate over our available translation files in the src/_data/landingpage/* directory, set a dynamic permalink (so "en" translations will be at /index.html and "fr" translations will be at /fr/index.html), then finally set some global `locale` and `title` properties which will be used for the `<html lang="{{ locale }}">` and `<title>{{ title }}</title>` tags.

---

```sh
tree -aI node_modules -I .git
.
├── .eleventy.js
├── .gitignore
├── package-lock.json
├── package.json
├── src/
│   ├── _data/
│   │   ├── about/
│   │   │   ├── en.json
│   │   │   └── fr.json
│   │   └── landingpage/
│   │       ├── en.json
│   │       └── fr.json
│   ├── _includes/
│   │   └── layouts/
│   │       ├── about.njk
│   │       ├── base.njk
│   │       └── landingpage.njk
│   ├── about.njk
│   └── landingpage.njk
└── www/
    ├── about/
    │   ├── fr/
    │   │   └── index.html
    │   └── index.html
    ├── fr/
    │   └── index.html
    └── index.html

10 directories, 17 files
```
